packages/node/src/protocol/utils/get-outcome-increments.ts

```diff
@@ -4,24 +4,51 @@ import { Contract } from "ethers";
 import { One, Zero } from "ethers/constants";
 import { BaseProvider } from "ethers/providers";
 import { BigNumber, bigNumberify, defaultAbiCoder } from "ethers/utils";
+import { isArray } from "util";
 
 import { StateChannel } from "../../models";
 
-function computeEthTransferIncrement(outcome): [string, BigNumber] {
-  const decoded = defaultAbiCoder.decode(["tuple(address,uint256)[]"], outcome);
-
-  if (
-    !(
-      decoded.length === 1 &&
-      decoded[0].length === 1 &&
-      decoded[0][0].length === 2
-    )
-  ) {
-    throw new Error("Outcome function returned unexpected shape");
+function unwrap(arr: any) {
+  if (!isArray(arr)) {
+    throw new Error("oops");
   }
-  const [[[address, to]]] = decoded;
+  if (arr.length !== 1) {
+    throw new Error("oops");
+  }
+  return arr[0];
+}
+
+// todo(xuanji): convert to object
+function computeEthTransferIncrement(outcome): { [s: string]: BigNumber } {
+  const decoded0 = defaultAbiCoder.decode(
+    ["tuple(address,uint256)[]"],
+    outcome
+  );
+  const decoded1 = unwrap(decoded0);
+
+  if (!isArray(decoded1)) {
+    throw new Error("no!");
+  }
+
+  const ret = {} as any;
 
-  return [address, to];
+  for (const pair of decoded1) {
+    if (!(isArray(pair) && pair.length === 2)) {
+      throw new Error("no!");
+    }
+    const [address, to] = pair;
+    ret[address] = to;
+  }
+  return ret;
+}
+
+function anyNoneroValues(a: { [s: string]: BigNumber }): Boolean {
+  for (const k in a) {
```

Comment:
Many cases of single character variable names in this PR, I recommend being more descriptive.